docs/book/src/migration/guide.md

```diff
@@ -0,0 +1,223 @@
+
+- domain: tutorial.kubebuilder.io
+- group: batch
+- version: v1
+- kind: CronJob
+
+## Initialize a v2 Project
+
+Now that we initialize a v2 project, we'll need to first initialize the go
+modules.
+
+```bash
+go mod init tutorial.kubebuilder.io/project
+```
+
+Then we finish initializing the project with kubebuilder.
+
+```bash
+kubebuilder init --domain tutorial.kubebuilder.io
+```
+
+## Migrate APIs and Controllers
+
+Next, we'll re-scaffold out the API types and controllers. Since we want both,
+we'll say yes to both the API and controller prompts when asked what parts we
+want to scaffold:
+
+```bash
+kubebuilder create api --group batch --version v1 --kind CronJob
+```
+
+If using multiple groups, it will require some manual work. Please follow
+[this][TODO-multi-group-doc] for more details.
+
+### Migrate the APIs
+
+Now let's copy the API definition from `pkg/apis/batch/v1/cronjob_types.go` to
+`api/v1/cronjob_types.go`. We only need to copy the implementation of the `Spec`
+and `Status` fields.
+
+We can replace the `+k8s:deepcopy-gen:interfaces` marker with
+`+kubebuilder:object:root=true`.
+
+We don't need the following markers any more.
+
+```go
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+```
+
+Our API types should look like the following.
+
+```go
+// +kubebuilder:object:root=true
+
+// CronJob is the Schema for the cronjobs API
+type CronJob struct {...}
+
+// +kubebuilder:object:root=true
+
+// CronJobList contains a list of CronJob
+type CronJobList struct {...}
+```
+
+### Migrate the Controllers
+
+Now let's copy the controller reconciler code from
+`pkg/controller/cronjob/cronjob_controller.go` to
+`controllers/cronjob_controller.go`.
+
+- Copy the fields from `ReconcileCronJob` struct to `CronJobReconciler`.
+- Copy the reconciler code of `Reconcile` function.
+- Copy the [rbac related markers](/reference/markers/rbac.md) to the new file.
+- Copy the code under `func add(mgr manager.Manager, r reconcile.Reconciler) error`
+to `func SetupWithManager`.
+
+## Migrate the Webhooks
+
+If you don't have a webhook, you can skip this section.
+
+### Core Types and External CRDs
+
+If you are using webhook for kubernetes core types (e.g. Pods) or for an
+external CRD that is not owned by you, you can refer the
+[controller-runtime example for builtin types][builtin-type-example]
+and do something similar. Kubebuilder doesn't scaffold much for these cases, but
+you can use the library in controller-runtime.
+
+### Scaffold Webhooks for our CRDs
+
+Now let's scaffold the webhooks, we'll need to run the following command with
+`--defaulting` and `--programmatic-validation` flags as need.
+
+```bash
+kubebuilder create webhook --group batch --version v1 --kind CronJob --defaulting --programmatic-validation
+```
+
+Depending on how many CRDs need webhooks, we may need to run the above command
+multiple times with different Group, Version and Kind.
```

Comment:
oxford comma (`Group, Version, and Kind`), although this whole construction is a bit awkward with the plurality, so I'd use

```suggestion
multiple times with different Group-Version-Kinds.
```